res/games/gameitem.jinja

```diff
@@ -0,0 +1,120 @@
+            {% set player_login = "<font color='blue'>"~player_login~"</font>" %}
+        {% endif %}
+
+        <tr>
+        {% if teamalign == "left" -%}
+            <td align='right' width='{{ width }}'>
+            {{ player_login }}
+            </td>
+            <td>
+            <img src='{{ player_country }}'>
+            </td>
+            <td align='right' width='10'>
+            ({{ player_global_rating }})
+            </td>
+        {% else %}
+            <td align='left' width='10'>
+            ({{ player_global_rating }})
+            </td>
+            <td>
+            <img src='{{ player_country }}'>
+            </td>
+            <td align='left' width={{ width }}>
+            {{ player_login }}
+            </td>
+        {% endif %}
+         </tr>
+{% endmacro -%}
+
+
+{%- macro versus_string(fontsize) %}
+    <td valign='middle'>
+    <font color ='black' size={{ fontsize }}>VS</font>
+    </td>
+{% endmacro -%}
+
+
+{%- macro modtip(mods) %}
+    <b><br/>With:</b><br/>
+    {{"<br />".join(mods)}}
+{% endmacro -%}
+
+
+{%- macro observertooltip(observer) %}
+     {% if observer.login is defined %}
+        {% set observer_country = iconpath~observer.country~".png" %}
+        {% set observer_global_rating = observer.rating_estimate %}
+    {% endif %}
+    {% set observer_login = observer.login %}
+    {% if observer_login is not defined %}
+        {% set observer_login = observer %}
+    {% endif %}
+    <br/>{{ observer_login }} <img src='{{ observer_country }}'> ({{ observer_global_rating }})
+{% endmacro -%}
+
+
+{%- macro TooltipFormatting(teams, mods, observers, me, iconpath) %}
+    <table border="0" cellpadding="0" cellspacing="5">
+    <tbody>
+    <tr>
+        {% for team in teams %}
+            {% if loop.last %}
+                <td><table>
+                    {% for player in team %}
+                        {{ playermacro(player, me, iconpath, teamalign="right") }}
+                    {% endfor %}
+                </table><td>
+            {% else %}
+                <td><table>
+                    {% for player in team %}
+                        {{ playermacro(player, me, iconpath, teamalign="left") }}
+                    {% endfor %}
+                </table><td>
+                {{ versus_string(fontsize='+{}'.format(1 + (((teams[0])|length + (teams[1])|length)/2)|int)) }}
+            {% endif %}
+        {% endfor %}
+    </tr>
+        {% if observers %}
+            <tr>
+            <td>
+            <b>Observers:</b>
+            {% for observer in observers %}
+                <span style='white-space: nowrap;'>{{ observertooltip(observer) }}</span>
+            {% endfor %}
+            </td>
+            </tr>
+        {% endif %}
+        {% if mods %}
+            <tr>
+            <td>
+            {{ modtip(mods) }}
+            </td>
+            </tr>
+        {% endif %}
+{% endmacro -%}
+
+{% if teams[0] is defined or observers is defined %}
```

Comment:
Do things still break when the team list is empty / None? If so, then we should fix that in the class that prepares the tooltip data.

Reply:
well things dont break because of the checks - will an addition in python change the outcome though? i dont know what id have to fix in the tooltip data class

Reply:
i could not reproduce while at home. it may be related to my slow internet on work after all - the latency is probably very high there.